cmake/modules/CheckCompileTargets.cmake

```diff
@@ -0,0 +1,70 @@
+# CMake module to get the compile options to match the target 
```

Comment:
@tchaikov Kefu, thanks a lot! Tow commits can make the PR clearer. I will do it now.
Also I will correct the commit message. Thanks!